### ISSUE TITLE
Can't remove sash from between splitter on Windows.

### ISSUE THREAD

**Issue by KraziMeanieDev** [wxWidgets/wxWidgets#24966]
#### Bug description:
When using wxSplitterWindow to create a splitter between two panels, an obtrusive sash is rendered that can't be visually removed. 

#### Expected vs observed behaviour: 
When wxSP_NOBORDER OR wxSP_BORDER OR whatever control to remove the Sash/Border is used, it should be removed. But that doesn't seem to be the case.

#### To Reproduce:
```
#include <wx/splitter.h>
#include <wx/wx.h>

class TestApp : public wxApp {
public:
	bool OnInit() override;
};

class AppWindow : public wxFrame {
public:
	AppWindow(const wxString &title, const wxPoint &pos, const wxSize &size);

private:
	void OnExit(wxCommandEvent &event);
};

bool TestApp::OnInit()
{
	AppWindow *window = new AppWindow("Hello WxTest", wxDefaultPosition, wxDefaultSize);
	window->SetSize(1280, 720);
	window->Show(true);
	return true;
}

AppWindow::AppWindow(const wxString &title,
	const wxPoint &pos,
	const wxSize &size)
	: wxFrame(nullptr, wxID_ANY, title, pos, size)
{
	wxSplitterWindow *splitter = new wxSplitterWindow(this,
		wxID_ANY,
		wxDefaultPosition,
		wxDefaultSize,
		wxSP_NOBORDER | wxSP_LIVE_UPDATE);

	wxPanel *sidebar = new wxPanel(splitter);
	sidebar->SetBackgroundColour(wxColor(2, 6, 23));
	// Creates a WxPanel called window and set's its background color;
	wxPanel *editor = new wxPanel(splitter);
	editor->SetBackgroundColour(wxColor(15, 23, 42));

	splitter->SetMinimumPaneSize(200); // Matches sidebar
	splitter->SplitVertically(sidebar, editor);
};

void AppWindow::OnExit(wxCommandEvent &event)
{
	Close(true);
};

wxIMPLEMENT_APP(TestApp);
```

Image: ![image](https://github.com/user-attachments/assets/de037275-9fbe-4cbd-a460-a23e8d11a615)

### Platform and version information

- wxWidgets version  you use: 3.2.6
- wxWidgets port  you use: wxMSW
- OS  and its version: Windows 11

**Comment by KraziMeanieDev** [comment]
WxSP_NOSASH hides the sash but makes it unusable meaning but make it unusable by making it it unable to resize the spiltter.

**Comment by vadz** [comment]
Yes, currently you either show the splitter and allow using it or don't show it — and then it can't be used. I don't see how can completely removing it and still allow to move it could work.

**Comment by KraziMeanieDev** [comment]
Okay maybe I'm not explaining myself properly. The dash is extremely visible. It's there a way to bring to two panels closer together? Like setting the size of the sash? I tried using the SetSashSize but that's been deprecated. Is there an alternative method?

**Comment by vadz** [comment]
> Okay maybe I'm not explaining myself properly.

I've indeed misunderstood you because you had written "sash can't be visually removed", so I thought you wanted to remove it completely.

> The dash is extremely visible. It's there a way to bring to two panels closer together? Like setting the size of the sash? I tried using the SetSashSize but that's been deprecated. Is there an alternative method?

No because the sash is drawn by the native theme. I guess we could undeprecate `SetSashSize()` and draw some "generic" sash using the code from `wxRendererGeneric::DrawSplitterSash()` after extending it to take a parameter instead of hardcoding 3 for the sash width. But someone would need to do it because I'm personally not interested in using anything other than the native-looking sash.

**Comment by KraziMeanieDev** [comment]
Ah okay that makes a lot of sense. Thanks so much for the help. From the looks of it I guess I'm just using the wrong tool for the job? I'm creating my own ide as a project to further learn c++ and I want to an editor, file tree, terminal layout. I'm seeing a lot of different layout option, but none are clear on how to achieve this. I figured splitter would satisfy this functionality, but it seems like that's not the case. As for the non-native looking sash implementation, I guess I'll take a jab at it.

**Comment by KraziMeanieDev** [comment]
Ahh nevermind I realize that I was referencing Visual Studio Code to get a visual on how it should look and work but after taking a look at Visual Studio, Window Explorer and QT Creator I have a better understanding of how it should look and how its implemented. I'll just have to make it look better. Although I'm seeing that in msw/renderer.cpp there's `static const wxCoord SASH_WIDTH = 4; // the width of the sash: this is the same as used by Explorer...`  but it seems like the one in explorer is actually much smaller than that right?

![image](https://github.com/user-attachments/assets/61d74977-f1d0-4dab-bc8d-c701b7f017f9)

**Comment by oneeyeman1** [comment]
@KraziMeanieDev ,
Are you working on the DarkMode? Can you try the light mode?

Thx.

**Comment by KraziMeanieDev** [comment]
@oneeyeman1 What do you mean? IF my system is using dark mode? Yeah, it is. IF you're talking about the application itself, isnt a dark mode just dark colors.

**Comment by oneeyeman1** [comment]
@KraziMeanieDev ,
What is meant is for you to try and run your app or a splitter sample in the light or normal mode. 

Thank you.

**Comment by vadz** [comment]
Now that I think about it, you could define your own `wxRendererNative` (well, you'd actually inherit from `wxDelegateRendererNative`) and override its `GetSplitterParams()` and `DrawSplitterSash()` to draw your custom sash.

**Comment by KraziMeanieDev** [comment]
@oneeyeman1 The app is already being run in light mode. As per the documentation "Under MSW, the default appearance is always light and the applications that want to follow the system appearance need to explicitly call this function with Appearance::System parameter in order to do it."
